packages/cape/src/methods.ts

```diff
@@ -79,7 +87,7 @@ export abstract class Methods {
   /**
    * Connect to the Cape server.
    */
-  public async connect({ id }: ConnectArgs): Promise<void> {
+  public async connect({ id, functionCheckSum }: ConnectArgs): Promise<void> {
```

Comment:
You shouldn't need to pass the `functionCheckSum` here if you already initialized it in the `Cape` constructor and have the method to access it. Create the abstract method and you'll be able to access the value:

```
 public abstract getFunctionCheckSum(): string | undefined;
```

Reply:
So in run, how would I call this function? I also think that Cape constructor should not take a function token or a functionChecksum because those parameters are function specific.

Reply:
Each Cape instance connects to a single function so all connection details are tied to the Cape initialization. If you wanted to access the checksum, inside `run` you could call `this.getFunctionCheckSum()` to get the value.

Reply:
Gotcha, if this is the case then maybe ID should also be a part of Cape initialization?

Reply:
It might have been put there since it was originally the authToken and that was considered global. Perhaps it should have been put on the connect method like you have here.